rust/signed_doc/src/validator/ver.rs

```diff
@@ -0,0 +1,53 @@
+//! Validator for Signed Document Version
+
+use std::time::{Duration, SystemTime};
+
+use crate::{providers::CatalystSignedDocumentProvider, CatalystSignedDocument};
+
+pub(crate) struct VerRule;
+
+impl VerRule {
+    pub(crate) fn check<Provider>(
+        &self,
+        doc: &CatalystSignedDocument,
+        _provider: &Provider,
+    ) -> anyhow::Result<bool>
+    where
+        Provider: CatalystSignedDocumentProvider,
+    {
+        let id = doc.doc_id().ok();
```

Comment:
you could simplify this code in the following way.
```rust
let Ok(id) = doc.doc_id() else {
  doc.report().missing_field(
                "id",
                "Can't get a document id during the validation process",
            );
 return false;
}

let Ok(ver) = doc.doc_ver() else {
  doc.report().missing_field(
                "ver",
                "Can't get a document ver during the validation process",
            );
 return false;
}
```
In this scenario its fine to early return, because we need to have `id` and `ver` fields and we can't proceed without them, and its mandatory to have them in general, so even during decoding process we will catch that they are both missing.